### ISSUE TITLE
`-Wformat=` GCC 12 compiler warnings (Windows)

### ISSUE THREAD

**Issue by vszakats** [ngtcp2/nghttp3#173]
```
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_map.c: In function 'nghttp3_map_print_distance':
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_map.c:144:21: warning: unknown conversion type character 'z' in format [-Wformat=]
  144 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                     ^~~~~~~~~~~~~~~~~~~~~
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_map.c:144:58: note: format string is defined here
  144 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                                                          ^
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_map.c:144:21: warning: unknown conversion type character 'z' in format [-Wformat=]
  144 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                     ^~~~~~~~~~~~~~~~~~~~~
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_map.c:144:71: note: format string is defined here
  144 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                                                                       ^
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_map.c:144:21: warning: too many arguments for format [-Wformat-extra-args]
  144 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                     ^~~~~~~~~~~~~~~~~~~~~

/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_ksl.c: In function 'ksl_print':
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_ksl.c:731:24: warning: unknown conversion type character 'z' in format [-Wformat=]
  731 |   fprintf(stderr, "LV=%zu n=%u\n", level, blk->n);
      |                        ^
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_ksl.c:731:30: warning: format '%u' expects argument of type 'unsigned int', but argument 3 has type 'size_t' {aka 'long
  731 |   fprintf(stderr, "LV=%zu n=%u\n", level, blk->n);
      |                             ~^     ~~~~~
      |                              |     |
      |                              |     size_t {aka long long unsigned int}
      |                              unsigned int
      |                             %llu
/home/runner/work/curl-for-win/curl-for-win/nghttp3/lib/nghttp3_ksl.c:731:19: warning: too many arguments for format [-Wformat-extra-args]
  731 |   fprintf(stderr, "LV=%zu n=%u\n", level, blk->n);
      |                   ^~~~~~~~~~~~~~~
```
Ref: https://github.com/curl/curl-for-win/actions/runs/6711735060/job/18239769146#step:3:3828

**Comment by tatsuhiro-t** [comment]
Fixed by #174.

**Comment by vszakats** [comment]
Thank you @tatsuhiro-t!
